boards: stm32h7s78_dk: add FDCAN1 support

bklaric1 opened the pull request:
## Changes
- Added FDCAN1 configuration to `stm32h7s78_dk-common.dtsi` and disabled it by default. Also explained that in order to use FDCAN1, the user needs to disable `i2c1`, since they share the `pb9` pin.
- ~Updated `stm32h7s78_dk.yaml` and `stm32h7s78_dk_stm32h7s7xx_ext_flash_app.yaml` to include the CAN feature.~
- Updated `index.rst` to include the CAN feature and documented the steps needed in order to use it.

## Tests
I tested these changes successfully with the following example: [can/counter](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/can/counter)



## Notes
Closes #99752

bklaric1 commented on boards/st/stm32h7s78_dk/doc/index.rst:
```diff
@@ -200,6 +201,13 @@ USB
+-----
+
+STM32H7S78-DK Discovery board has 2 FDCAN bus interfaces. Currently, only
+FDCAN1 is supported. It is not enabled by default. To enable it, make sure
```
Done. Had trouble with the wording for this one. Thanks.

djiatsaf-st reviewed (commented):
Could you please also test or provide support for the following tests?

- tests/drivers/can/api
- tests/drivers/can/shell
- tests/drivers/can/timing

These are usually the main tests we use to validate CAN peripherals.

henrikbrixandersen reviewed (changes requested):
I suggest leaving out `can` as a supported feature (since CAN is not enabled in the default board configuration) and thus also leaving out the test overlays (as they will never be used except in your own testing efforts).

djiatsaf-st commented on boards/st/stm32h7s78_dk/stm32h7s78_dk.yaml:
```diff
@@ -16,4 +16,5 @@ supported:
   - octospi
   - usbd
   - memc
+  - can
```
> @djiatsaf-st answered this indirectly with his [comment](https://github.com/zephyrproject-rtos/zephyr/pull/99917#pullrequestreview-3504550573) if I am not mistaken.

From my point of view, adding the overlays would be better, and therefore, removing "can" from stm32h7s78_dk.yaml makes sense since there is not enough flash on this variant.

The goal is to run the tests on the H7S_DK board and ensure all tests work well. It should not be just about adding the overlays.

I suspect there is insufficient flash to perform these tests. Rather than adding the overlay for stm32h7s78_dk, I would prefer to add stm32h7s78_dk_stm32h7s7xx_ext_flash_app.overlay for the **can/timing and can/api** tests on this variant.

bklaric1 commented on tests/drivers/can/shell/boards/stm32h7s78_dk.overlay:
Okay. I will amend my last commit to remove the added tests and tag.

etienne-lms reviewed (changes requested):
Could you squash the 2 commits?
I can remove my 'Request changes' feedback if @erwango and @henrikbrixandersen request so, but I think it's a bit ugly to have a commit right followed by a fixup change.

bklaric1 commented:
Okay, no problems. I even offered to do it initially.
I will squash 2. commit into 1. commit.

FRASTM reviewed (commented):
very minor remark
LGTM

bklaric1 committed:
boards: stm32h7s78_dk: add FDCAN1 support

Add FDCAN1 configuration to the board's .dtsi file and
comments explaining how to enable it, since fdcan1 and i2c1
share the pb9 pin. Updated documentation to explain CAN usage.

Signed-off-by: Benjamin Klaric <benjamin.klaric01@gmail.com>
